### ISSUE TITLE
Generic types for SlashCommand Options

### ISSUE THREAD

**Issue by joaisa17** [discordjs/discord.js#8749]
### Which package is the feature request for?

discord.js

### Feature

I'm obsessed with autocomplete, and would like to pass an interface into any of the SlashCommand Option classes, to provide autocomplete when adding e.g choices:

Suggested usage:
```ts
import { SlashCommandBuilder, SlashCommandStringOption, SlashCommandIntegerOption } from 'discord.js';

interface Options {
    somestring: string;
    somestringchoice: 'foo'|'bar';

    somenumber: number;
}

new SlashCommandBuilder()
    .setName('example')
    .setDescription('Example command')

    .addStringOption(
        new SlashCommandStringOption<Options>()
            .setName('somestring') // Provide autocomplete for keyof Options (filter string types)
            .setDescription('Any string')
    )

    .addStringOption(
        new SlashCommandStringOption<Options>()
            .setName('somestringchoice') // Provide autocomplete for keyof Options (filter literal string types if possible)
            .setDescription('String choice')
            .setChoices(
                { name: 'foo', value: 'foo' }, // Provide autocomplete for Options[name]
                { name: 'bar', value: 'bar' }
            )
    )

    .addIntegerOption(
        new SlashCommandIntegerOption<Options>()
            .setName('somenumber') // Provide autocomplete for keyof Options (filter number types)
            .setDescription('Number choice')
    )
```

### Ideal solution or implementation

Add an optional generic to each of the Option classes

### Other context

Adding a generic to the SlashCommandBuilder class itself would probably mean that you will have to rewrite the entire usage of the SlashCommandBuilder, as I do not believe a passed class can use generics from another, e.g you pass the Options to the Builder itself, in which case the StringOption class won't resolve the options.

**Comment by RedGuy12** [comment]
Duplicate of #8218

**Comment by imranbarbhuiya** [comment]
This isn't a duplicate of #8218 as that issue is for getting type in received values, not in the builder methods. And I don't see why you want a autocomplete in builders. Both interface and builders are defined by you so why you want to duplicate your codes with a interface as well as builder

**Comment by didinele** [comment]
We're currently working on builders so it's probably a good time to take a look at this.

I'm personally leaning against no here. The autocomplete pro isn't that big, and thinking about actual type-safety, this seems like it could be useful *sometimes*? cc @vladfrangu

**Comment by almeidx** [comment]
After #10448 and an internal discussion, we decided not to go forward with this
